services/matrix/default.nix

```diff
@@ -0,0 +1,162 @@
+{ lib, config, pkgs, baseDomain, ... }:  let
+  matrixPort = 8008;
+  isDev = (builtins.substring 0 3 baseDomain) == "dev";
+  synapseDb = config.services.matrix-synapse.settings.database.args;
+  initSynapseDb = ''CREATE DATABASE "${synapseDb.database}" WITH OWNER "${synapseDb.user}" ENCODING "UTF8" TEMPLATE template0 LC_COLLATE = "C" LC_CTYPE = "C";'';
```

Comment:
Is encoding UTF8 correct here?

Reply:
I took a look at the database on hopper (current prod matrix server, deployed from ansible) and it is currently on C collate:

```
postgres=# select datname, datcollate FROM pg_database WHERE datname = 'synapse';
 datname | datcollate 
---------+------------
 synapse | C
(1 row)
```

The encoding is set to UTF8:

```
postgres=# SELECT pg_encoding_to_char(encoding) FROM pg_database WHERE datname = 'synapse';
 pg_encoding_to_char 
---------------------
 UTF8
```

Reply:
Interesting.